chainer/variable.py

```diff
@@ -52,6 +52,52 @@ def make_message(message):
         raise ValueError(make_message(msg))
 
 
+def variable_repr(var):
+    """Return the string representation of a variable.
+
+    Args:
+        var (~chainer.Variable): Input Variable.
+    .. seealso:: numpy.array_repr
+    """
+    xp = cuda.get_array_module(var)
+    if xp is numpy:
+        arr = var.data
+    else:
+        arr = var.data.get()
+
+    if arr.size > 0 or arr.shape == (0,):
+        if var.name:
+            prefix = 'variable ' + var.name + '('
+        else:
+            prefix = 'variable('
+        lst = numpy.array2string(arr, None, None, None, ', ', prefix)
+    else:  # show zero-length shape unless it is (0,)
+        lst = '[], shape=%s' % (repr(arr.shape),)
+        if var.name:
+            lst = var.name + lst
+    return '%s(%s)' % (prefix[:-1], lst)
```

Comment:
Or, as for `prefix[:-1]`, it could also be reasonable to leave `'('` in `prefix` and `return prefix + lst + ')'`.
In this way it will be similar to the formatting method in `variable_str()`.